packages/mobile/src/web3/contracts.ts

```diff
@@ -98,54 +114,88 @@ export function* initContractKit() {
 export function destroyContractKit() {
   Logger.debug(`${TAG}@closeContractKit`)
   contractKit = undefined
-  gethWallet = undefined
-  ipcProvider = undefined
+  wallet = undefined
 }
 
-export function* getContractKit() {
-  if (!contractKit) {
-    yield call(initContractKit)
+let initContractKitLock = false
+
+function* waitForContractKit(tries: number) {
```

Comment:
Why create a generator for this? Seems like it just dupes the existing waitForContractKitAsync. Note, async functions can be called from generators.

Reply:
Yup you're right I had a bit of a confusion there and was under the impression that there's an `async generator` thing that needs you happen but you can just `yield call(asyncFn)` 🤦 - will fix